src/ddmd/dsymbol.d

```diff
@@ -2007,6 +2029,78 @@ extern (C++) final class OverloadSet : Dsymbol
 }
 
 /***********************************************************
+ * Forwarding ScopeDsymbol.
+ * This is used by ForwardingAttribDeclaration and ForwardingScopeDeclaration
+ * to forward symbol insertions to another scope.
+ */
+extern (C++) final class ForwardingScopeDsymbol : ScopeDsymbol
+{
+    ScopeDsymbol forward;
+    extern (D) this(ScopeDsymbol forward)
+    {
+        super(null);
+        this.forward = forward;
+    }
+    override Dsymbol symtabInsert(Dsymbol s)
+    {
+        assert(!!forward);
+        if (auto d = s.isDeclaration())
+        {
+            if (d.storage_class & STClocal)
+            {
+                if (!symtab)
+                {
+                    symtab = new DsymbolTable();
+                }
+                return super.symtabInsert(s);
+            }
+        }
+        if (!forward.symtab)
+        {
+            forward.symtab = new DsymbolTable();
+        }
+        return forward.symtabInsert(s);
+    }
+
+    override Dsymbol symtabLookup(Dsymbol s,Identifier id)
+    {
+        assert(!!forward);
+        if (auto d = s.isDeclaration())
+        {
+            if (d.storage_class & STClocal)
```

Comment:
`STClocal` declarations are inserted into the inner scope (hence need to use `super.symtabInsert`). This is necessary because we do not want the `foreach` loop variable to clash with itself. Other declarations are forwarded (those use `forward.symtabInsert`). I'll add comments.